flottform/server/src/database.ts

```diff
@@ -25,8 +26,46 @@ function createRandomEndpointId(): string {
 
 class FlottformDatabase {
 	private map = new Map<EndpointId, EndpointInfo>();
+	private cleanupTimeoutId: NodeJS.Timeout | null = null;
+	private cleanupPeriod: number;
+	private entryTTL: number; // Time-to-Live for each entry
+
+	constructor(cleanupPeriod = 30 * 60 * 1000, entryTTL = 25 * 60 * 1000) {
+		this.cleanupPeriod = cleanupPeriod;
+		this.entryTTL = entryTTL;
+		this.startCleanup();
+	}
+
+	private startCleanup() {
+		this.cleanupTimeoutId = setTimeout(this.cleanupFn.bind(this), this.cleanupPeriod);
+	}
+
+	private cleanupFn() {
+		if (!this.map || this.map.size === 0) return;
```

Comment:
I think if we fall down to size 0, it will never call the `cleanupFn` again. Is there a test for this?

Reply:
No. I have to add one

Reply:
The cleanup mechanism is designed to run periodically, regardless of whether the map is empty at the time of execution. The purpose is to ensure stale entries are regularly deleted, even if none exist during a specific cleanup cycle. This mechanism continues to run until the server hosting the database is stopped.

For example, while the map might be empty during the first cleanup call, it’s possible that by the next cycle, new entries have been added, with some of them becoming stale. Stopping the cleanup when the map is empty will lead to not cleaning up stale entries in subsequent cycles. Therefore, it’s crucial that the cleanup function runs unconditionally.